harness/determined/profiler.py

```diff
@@ -281,51 +285,11 @@ def _end_collection(self) -> None:
                 self.timings_batcher_thread.send_shutdown_signal()
                 self.timings_batcher_thread.join()
 
-            self.sender_thread.send_shutdown_signal()
             self.sender_thread.join()
 
             self.has_finished = True
 
 
-class Timing:
```

Comment:
IMO we should keep Timings or something similar so it is easy to give users a well-defined way to provide their own timings without relying on our context manager - particularly as we add non-PyTorch frameworks or let users add their own as generic APIs

Reply:
Yeah I agree we'll need a nice user facing API, we could probably have an extended discussion about the right API, but I'm not sure if we should try to have it here since this is not user facing yet. I was just trying to save some LOC. If you still want this I can add it back.